cmd/main.go

```diff
@@ -145,7 +168,43 @@ func parseCliArgs(flags *args) (*cobra.Command, error) {
 	return rootCmd, nil
 }
 
-func run(kubeconfigArgs *genericclioptions.ConfigFlags, flags args, printFlags *k8sget.PrintFlags) (int, error) {
+func loadConfig(configPath string) (conf Config, err error) {
+	viper.SetConfigFile(configPath)
+	viper.SetConfigType("yaml")
+
+	viper.AutomaticEnv()
+
+	err = viper.ReadInConfig()
+	if err != nil {
+		return
+	}
+
+	err = viper.Unmarshal(&conf)
+	return
+}
+
+func validateConfig(conf Config) error {
+	for i, exclusion := range conf.Exclusions {
+		if exclusion.Description == nil {
+			return fmt.Errorf("description must not be empty for exlusion rule %d", i)
+		}
```

Comment:
I don't see any advantage having a config field description and even if it should not be a must field.
description can just be commented in the yaml if somebody wants to add some hints,

Reply:
it's printed out in the debug session to know which rule we matched, so it might be at least a name ?
cf. https://github.com/darkweaver87/gitops-zombies/blob/feat/filter-config/pkg/collector/resource.go#L179